signal/src/circular_buffer.cc

```diff
@@ -0,0 +1,290 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "signal/src/circular_buffer.h"
+
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define ASSERT assert
```

Comment:
Any reason we can't use the TFLITE_DCHECK macros for asserts here?

Reply:
no idea , all of this was written by Speech Team from my understanding they might have a reason I'm not sure of. Probably good question for Shlomi. I'd prefer to not change this till all the code is ported and we can see if it breaks anyone.